### PR TITLE
Make Cloud Spanner Mutations serializable

### PR DESCRIPTION
**Pull request by mairbek**
Same goal as #1939 but uses standard Java serialization and looks much cleaner.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/MutationTest.java [hunk 1]
```diff
@@ -187,7 +193,7 @@ public void equalsAndHashCode() {
 
     // Equality, not identity.
     tester.addEqualityGroup(
-        Mutation.newInsertBuilder("T1").build(), Mutation.newInsertBuilder("T1").build());
+            Mutation.newInsertBuilder("T1").build(), Mutation.newInsertBuilder("T1").build());
```
Comment (vkedia) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/MutationTest.java]: why are there so many formatting changes in this PR?

**Comment by vkedia** [comment]
Added @garrettjonesgoogle since you are touching some of the common classes.

**Comment by mairbek** [comment]
Added more tests and restored formatting

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/MutationTest.java (hunk 1 above)
Comment (mairbek) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/MutationTest.java]: Reverted formatting.

**Comment by garrettjonesgoogle** [comment]
The google-cloud-core changes look good.

**Pull request merged**